src/leader.py

```diff
@@ -152,6 +163,62 @@ def lead(self):
             self.turn(code, position)
 
 
+    def demo_circle(self):
+        """
+        Controls the leader vehicle in a circle for demo purposes.
+
+        Drives the leader vehicle in a continuous circle till script is
+        terminated.
+        """
+        while True:
+            self.fw.turn(DEMO_LEFT_TURN)
+            self.bw.speed(DEMO_SPEED)
+            self.bw.forward()
+
+
+    def demo_line(self):
+        """
+        Controls the leader vehicle in a line for demo purposes.
+
+        Drives the leader vehicle in a line then turns around and repeats till
+        script is terminated.
+        """
+        while True:
+
+            straight_timer = time.time()
+            while time.time - straight_timer < DEMO_STRAIGHT_TIME:
```

Comment:
time.time()

Reply:
We don't need this to be in a while loop. Just do those three things, then wait. Otherwise we're sending commands to the wheels over and over again for no reason.